metrics/bucket.go

```diff
@@ -0,0 +1,51 @@
+package metrics
+
+import (
+	"context"
+
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
+)
+
+// NewBucketCounter creates a Float64Counter with units of bytes and a default set of aggregation
+// bounds from .25 kilobytes to 10 megabytes.
+func NewBucketCounter(name, desc string, tagKeys ...tag.Key) *Float64Counter {
+	defaultBounds := []float64{250, 500, 1000, 5000, 10000, 50000, 100000, 500000, 1000000, 5000000, 10000000}
+	return NewBucketCounterWithBuckets(name, desc, stats.UnitBytes, defaultBounds, tagKeys...)
+}
+
+// NewBucketCounterWithBuckets creates a Float64Counter wrapping an opencensus int64 measurement.
+func NewBucketCounterWithBuckets(name, desc, unit string, bounds []float64, tagKeys ...tag.Key) *Float64Counter {
+	log.Infof("registering avg counter: %s - %s", name, desc)
+	measure := stats.Float64(name, desc, unit)
+	iView := &view.View{
+		Name:        name,
+		Measure:     measure,
+		Description: desc,
+		TagKeys:     tagKeys,
+		Aggregation: view.Distribution(bounds...),
+	}
+	if err := view.Register(iView); err != nil {
+		// a panic here indicates a developer error when creating a view.
+		// Since this method is called in init() methods, this panic when hit
+		// will cause running the program to fail immediately.
+		panic(err)
+	}
+
+	return &Float64Counter{
```

Comment:
Dumb question: why do we need float64s to track number of bytes in open census?

Reply:
Not a dumb question, this is wrong and I have revised its definition in the recent changes
(same part of my vacation brain that wrote "con-snuu-suu" wrote this)